domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt

```diff
@@ -161,101 +192,305 @@ class TopicListController @Inject constructor(
+    storyId: String,
+    story: StorySummary,
+    mostRecentCompletedChapterProgress: ChapterProgress,
+    completedChapterProgressList: List<ChapterProgress>,
+    topic: Topic,
+    completedStoryTopicId: String
+  ): Pair<PromotedStory?, Long> {
+    var numberOfDaysPassed = 0L
+    val lastChapterSummary: ChapterSummary? =
+      story.chapterList.find { chapterSummary ->
+        mostRecentCompletedChapterProgress.explorationId == chapterSummary.explorationId
+      }
+    val nextChapterIndex = story.chapterList.indexOf(lastChapterSummary) + 1
+    if (story.chapterList.size > nextChapterIndex) {
+      val nextChapterSummary: ChapterSummary? = story.chapterList[nextChapterIndex]
+      if (nextChapterSummary != null) {
+        numberOfDaysPassed = mostRecentCompletedChapterProgress.getNumberOfDaysPassed()
+        return Pair(
+          addPromotedStoryInRecommendedStoryList(
+            storyId,
+            topic,
+            completedChapterProgressList.size,
+            story.chapterCount,
+            nextChapterSummary.name,
+            nextChapterSummary.explorationId,
+            completedStoryTopicId
+          ),
+          numberOfDaysPassed
+        )
+      }
     }
-    return ongoingStoryListBuilder.build()
+    return Pair(null, numberOfDaysPassed)
   }
 
-  private fun createRecommendedStoryList(): List<PromotedStory> {
-    val recommendedStories = ArrayList<PromotedStory>()
+  private fun addPromotedStoryInRecommendedStoryList(
+    storyId: String,
+    topic: Topic,
+    completedChapterProgressListSize: Int,
+    chapterCount: Int,
+    chapterSummaryName: String,
+    chapterSummaryExplorationId: String,
+    completedStoryTopicId: String
+  ): PromotedStory {
+    return createPromotedStory(
+      storyId,
+      topic,
+      completedChapterProgressListSize,
+      chapterCount,
+      chapterSummaryName,
+      chapterSummaryExplorationId,
+      completedStoryTopicId
+    )
+  }
+
+  private fun ChapterProgress.getNumberOfDaysPassed(): Long {
+    return TimeUnit.MILLISECONDS.toDays(
+      oppiaClock.getCurrentCalendar().timeInMillis - this.lastPlayedTimestamp
+    )
+  }
+
+  private fun computeSuggestedStories(
+    topicProgressList: List<TopicProgress>
+  ): List<PromotedStory> {
+    val recommendedStories = mutableListOf<PromotedStory>()
+
     val topicIdJsonArray = jsonAssetRetriever
       .loadJsonFromAsset("topics.json")!!
       .getJSONArray("topic_id_list")
-    for (i in 0 until topicIdJsonArray.length()) {
-      createRecommendedStoryFromAssets(topicIdJsonArray[i].toString())?.let {
-        recommendedStories.add(it)
+
+    val topicIdList = (0 until topicIdJsonArray.length()).map { topicIdJsonArray[it].toString() }
+
+    val index = topicIdList.indexOf(topicProgressList.last().topicId)
+
+    for (i in (index + 1) until topicIdJsonArray.length()) {
+      if (topicIdJsonArray.length() > i &&
+        createRecommendedStoryFromAssets(topicIdJsonArray[i].toString()) != null
```

Comment:
This seems a bit wasteful--can we store the retrieved story in a val to avoid creating it twice?